### PR TITLE
{Batch} Fix race condition in test

### PR DESCRIPTION
**Pull request by cRui861**
**Description**
Fix https://github.com/Azure/azure-cli/issues/28500

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Batch

**Review by zhoxing-ms — DISMISSED** [review]

**Comment by zhoxing-ms** [comment]
@bebound Could you please help review this PR?

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py [hunk 1]
```diff
@@ -343,6 +343,10 @@ def test_batch_application_cmd(self, resource_group, storage_account):
 
 
 class BatchMgmtByosScenarioTests(ScenarioTest):
+
+    def __init__(self, method_name):
+        super().__init__(method_name, random_config_dir=True)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py:348]: `*arg` and `*kwargs` should pass through. See #25689.

**Comment by jiasli** [comment]
This issue should be addressed as soon as possible as it is causing more CI failures: https://dev.azure.com/azclitools/public/_build?definitionId=32&_a=summary&branchFilter=487%2C487%2C487%2C487

**Review by jiasli — CHANGES_REQUESTED** [review]
See comments

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py (hunk 1 above)
Comment (cRui861) [reply src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py]: Updated

**Review by zhoxing-ms — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py [hunk 2]
```diff
@@ -343,6 +343,10 @@ def test_batch_application_cmd(self, resource_group, storage_account):
 
 
 class BatchMgmtByosScenarioTests(ScenarioTest):
+
+    def __init__(self, method_name, *arg, **kwargs):
+        super().__init__(method_name, *arg, random_config_dir=True, **kwargs)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py:348]: No need to explicitly specify `method_name`. It is part of `*arg`.

```suggestion
    def __init__(self, *arg, **kwargs):
        super().__init__(*arg, random_config_dir=True, **kwargs)
```

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py:348]: Well, as this pattern is frequently used elsewhere:

https://github.com/Azure/azure-cli/blob/21b54b5a7a41865b3f8c65640186d40479461f66/src/azure-cli/azure/cli/command_modules/serviceconnector/tests/latest/test_webpp_connection_scenario.py#L23-L27

You can keep it for now.

**Review by jiasli — APPROVED** [review]

**Pull request merged**